### PR TITLE
Claim $XYZ airdrop reward

### PR DESCRIPTION
**Pull request by strenev**
In this PR I have:
- Added Airdrop reward section with adjusted airdrop amount
- Added modal with claim button calling the MerkleDistributor claim function

### CODE REVIEW

File: src/modules/yield-farming/components/pool-rewards/index.tsx [hunk 1]
```diff
@@ -41,6 +43,13 @@ const PoolRewards: React.FC = () => {
     return (sum ?? BigNumber.ZERO).plus(contract.potentialReward ?? BigNumber.ZERO);
   }, undefined);
 
+  const merkleDistributorData = yfPoolsCtx.merkleDistributor;
+  const isAirdropClaimed = merkleDistributorData?.isAirdropClaimed;
+  const adjustedAmount = merkleDistributorData?.adjustedAmount;
+
+  let airdropAmount;
+  !isAirdropClaimed ? (airdropAmount = BigNumber.from(adjustedAmount)) : BigNumber.from(0);
```
Comment (wfm-slava) [inline src/modules/yield-farming/components/pool-rewards/index.tsx:51]: May be it should be

`const airdropAmount = !isAirdropClaimed ? BigNumber.from(adjustedAmount) : BigNumber.from(0);`

File: src/web3/merkleDistributor.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        this.isAirdropClaimed = false;
+        this.claimIndex = -1;
+        this.claimAmount = undefined;
+        this.adjustedAmount = undefined;
+        this.emit(Web3Contract.UPDATE_DATA);
+      }
+
+      const airdropAccounts = airdropData.map(drop => ({
+        account: drop.address,
+        amount: BigNumber.from(BigNumber.from(drop.earnings.toString())),
```
Comment (wfm-slava) [inline src/web3/merkleDistributor.ts:33]: BigNumber.from is applied twice

File: src/web3/merkleDistributor.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      }));
+      this.claimIndex = airdropAccounts.findIndex(o => o.account === this.account);
+      this.claimIndex !== -1
```
Comment (wfm-slava) [inline src/web3/merkleDistributor.ts:36]: `this.claimAmount = this.claimIndex !== -1 ? this.getClaimAmount(this.account || '', airdropData) : undefined;`

File: src/web3/merkleDistributor.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        ? (this.claimAmount = this.getClaimAmount(this.account || '', airdropData))
+        : (this.claimAmount = undefined);
+      this.adjustedAmount = undefined;
+    });
+  }
+
+  getClaimAmount(address: string, airdropData: any[]): string {
+    return airdropData.find(e => e.address === address).earnings;
```
Comment (wfm-slava) [inline src/web3/merkleDistributor.ts:44]: there is posibility the item not to be found, so

airdropData.find(e => e.address === address) will return undefined and '.earnings' will crash app,

would be better to have like this

```
getClaimAmount(address: string, airdropData: any[]): string | undefined {
    return airdropData.find(e => e.address === address)?.earnings;
}
```

File: src/modules/yield-farming/components/pool-airdrop-modal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+const AirdropModal: FC<AirdropModalProps> = props => {
+  const { merkleDistributor, ...modalProps } = props;
+
+  const walletCtx = useWallet();
+
+  const [claiming, setClaiming] = useState(false);
+
+  const merkleDistributorContract = merkleDistributor;
+
+  const airdropAccounts = airdropData.map(drop => ({
```
Comment (wfm-slava) [inline src/modules/yield-farming/components/pool-airdrop-modal/index.tsx:31]: I would suggest to memo this

```
const tree = useMemo(() => {
  const airdropAccounts = airdropData.map(drop => ({
    account: drop.address,
    amount: BigNumber.from(drop.earnings.toString()),
  }));

  return new BalanceTree(airdropAccounts);
}, []);
```

**Review by wfm-slava — CHANGES_REQUESTED** [review]

File: src/web3/merkleDistributor.ts (hunk 4 above)
Comment (strenev) [reply src/web3/merkleDistributor.ts]: Done

File: src/web3/merkleDistributor.ts (hunk 3 above)
Comment (strenev) [reply src/web3/merkleDistributor.ts]: Done!

File: src/modules/yield-farming/components/pool-rewards/index.tsx (hunk 1 above)
Comment (strenev) [reply src/modules/yield-farming/components/pool-rewards/index.tsx]: also fixed

**Comment by strenev** [comment]
@wfm-slava @andreirykov guys, I've added the changes from the feedback

**Review by andreirykov — APPROVED** [review]

**Review by rizedr — APPROVED** [review]

**Pull request merged**